steve/scrapers.py

```diff
@@ -50,23 +48,14 @@ def scrape(self, url):
         if not is_youtube(url):
             return
 
-        # FIXME: Sometimes youtube-dl takes a *long* time to run. This
-        # needs to give indication of progress.
-        try:
-            output = subprocess.check_output(
-                ['youtube-dl', '-j', url],
-                stderr=subprocess.STDOUT
-            )
-        except subprocess.CalledProcessError as cpe:
-            raise ScraperError('youtube-dl said "{0}".'.format(cpe.output))
-        except OSError:
-            raise ScraperError('youtube-dl not installed or not on PATH.')
+        import youtube_dl
 
-        # Each line is a single JSON object.
-        items = []
-        for line in output.splitlines():
-            items.append(json.loads(line))
+        ydl = youtube_dl.YoutubeDL()
```

Comment:
You mean have a `ydl` as attribute of the `YoutubeScraper` instance?
I admit I haven't looked at how `YoutubeScraper` is used in steve